Sources/NIOHTTP1/HTTPDecoder.swift

```diff
@@ -486,8 +533,14 @@ public class HTTPDecoder<HTTPMessageT>: ByteToMessageDecoder, AnyHTTPDecoder {
         // be totally safe to call this with a null data pointer. Just to make sure, though,
         // let's store nil in the base address. Once we've called this, we've seenEOF: don't
         // let us enter this function again.
-        self.state.baseAddress = nil
         self.state.seenEOF = true
+
+        guard !self.decoding else {
+            // We are currently decoding, return early as we will handle it in the decoding loop.
```

Comment:
This comment is wrong: we don't handle the EOF in the decoding loop at all. We need to explicitly call `c_nio_http_parser_execute` with a length of `0`, which the other code never does. You may want to add a test that triggers this case to validate that it doesn't work, and then we can fix it. :smile: